src/Pages/Library/library.js

```diff
@@ -0,0 +1,74 @@
+import { useEffect } from "react"
+import axios from "axios"
+import { useAuth } from "../../Contexts/auth-context";
+import {Header} from "../../Components/header"
+import { useNavigate, Link  } from "react-router-dom";
+import "../Videolisting/videolisting.css"
+import { useServices } from "../../Contexts/service-context";
+
+const Library = () => {
+
+  const { videoState , dispatchVideo } = useServices()
+
+  const { 
+    auth: { token , status } ,
+  }  = useAuth()
+  const navigate = useNavigate()
+  
+  useEffect(() => {
+    axios.get("/api/user/likes" , { headers : {authorization : token}}).then((res) => {
+      dispatchVideo({type : "GET_LIKED_VIDEOS" , payload : res.data.likes})
+    })
+    axios.get("/api/user/watchlater" , {headers : {authorization : token}}).then(res => {
+      dispatchVideo({type : "GET_WATCHLATER_VIDEOS" , payload : res.data.watchlater})
```

Comment:
nothing wrong with this approach but as your app grows you will use these api calls many time so you should consider making a service-api file from where you would import these functions having async api calls